contracts/governance/extensions/GovernorOverrideDelegateVote.sol

```diff
@@ -0,0 +1,194 @@
+            return _countVotesOverride(proposalId, account, support, params);
+        }
+
+        totalWeight -= _overrideVoteWeight[account][proposalId];
+
+        ProposalVote storage proposalVote = _proposalVotes[proposalId];
+
+        if (proposalVote.voteReceipt[account].hasVoted) {
+            revert GovernorAlreadyCastVote(account);
+        }
+        proposalVote.voteReceipt[account] = VoteReceipt({hasVoted: true, support: support});
+
+        if (support == uint8(VoteType.Against)) {
+            proposalVote.againstVotes += totalWeight;
+        } else if (support == uint8(VoteType.For)) {
+            proposalVote.forVotes += totalWeight;
+        } else if (support == uint8(VoteType.Abstain)) {
+            proposalVote.abstainVotes += totalWeight;
+        } else {
+            revert GovernorInvalidVoteType();
+        }
+
+        return totalWeight;
+    }
+
+    function _countVotesOverride(
+        uint256 proposalId,
+        address account,
+        uint8 support,
+        bytes memory params
+    ) private returns (uint256) {
+        ProposalVote storage proposalVote = _proposalVotes[proposalId];
+        uint256 proposalSnapshot = proposalSnapshot(proposalId);
+        address delegate = _getPastDelegate(account, proposalSnapshot);
+
+        if (proposalVote.overrideVoteReceipt[account].hasVoted) {
+            revert GovernorAlreadyCastVoteOverride(account);
+        }
+
+        uint256 overrideWeight = _getPastBalanceOf(account, proposalSnapshot);
+
+        proposalVote.overrideVoteReceipt[account] = VoteReceipt({hasVoted: true, support: support});
+        if (support == uint8(VoteType.Against)) {
+            proposalVote.againstVotes += overrideWeight;
+        } else if (support == uint8(VoteType.For)) {
+            proposalVote.forVotes += overrideWeight;
+        } else if (support == uint8(VoteType.Abstain)) {
+            proposalVote.abstainVotes += overrideWeight;
+        } else {
+            revert GovernorInvalidVoteType();
+        }
+
+        // Account for the delegate's vote
+        VoteReceipt memory delegateVoteReceipt = proposalVote.voteReceipt[delegate];
+        if (delegateVoteReceipt.hasVoted) {
+            // If delegate has voted, remove the delegatee's vote weight from their support
+            if (delegateVoteReceipt.support == uint8(VoteType.Against)) {
+                proposalVote.againstVotes -= overrideWeight;
+            } else if (delegateVoteReceipt.support == uint8(VoteType.For)) {
+                proposalVote.forVotes -= overrideWeight;
+            } else if (delegateVoteReceipt.support == uint8(VoteType.Abstain)) {
+                proposalVote.abstainVotes -= overrideWeight;
+            } else {
+                revert GovernorInvalidVoteType();
+            }
+
+            // Write delegate into the params for event
+            assembly {
+                mstore(add(params, 0x20), or(mload(add(params, 0x20)), shl(64, delegate)))
+            }
```

Comment:
Not sure we should touch the params like that. There are many positions where another module could read them. If module start tampering with them, that may be confusing.